lib/ansible/module_utils/vmware.py

```diff
@@ -249,8 +262,10 @@ def gather_vm_facts(content, vm):
         facts['customvalues'][kn] = value_obj.value
 
     net_dict = {}
-    for device in vm.guest.net:
-        net_dict[device.macAddress] = list(device.ipAddress)
+    vmnet = _get_vm_prop(vm, 'vm.guest.net')
+    if vmnet:
+        for device in vm.guest.net:
```

Comment:
Not a necessary change but since the code just went through the trouble of retrieving that into the vmnet variable, might as well use it here:
``` python
for device in vmnet:
```

Looks slightly better to make use of it since you've retrieved it and a tiny bit faster (since python doesn't have to walk to the object's hierarchy to get the data structure that's being iterated over).